homeassistant/components/freebox/config_flow.py

```diff
@@ -0,0 +1,110 @@
+"""Config flow to configure the Freebox integration."""
+import logging
+
+from aiofreepybox.exceptions import AuthorizationError, HttpRequestError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import DOMAIN  # pylint: disable=unused-import
+from .router import get_api
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FreeboxFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize Freebox config flow."""
+        self._host = None
+        self._port = None
+
+    def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                    vol.Required(CONF_PORT, default=user_input.get(CONF_PORT, "")): int,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is None:
+            return self._show_setup_form(user_input, errors)
+
+        self._host = user_input[CONF_HOST]
+        self._port = user_input[CONF_PORT]
+
+        # Check if already configured
+        await self.async_set_unique_id(self._host)
```

Comment:
We have decided not to allow ip address in config entry unique id, since in some cases it could cause problems eg if two devices swaps ip addresses on the local network. Can the host be an ip address?

Reply:
The host is a DNS with this pattern `[ID].freeboxos.fr`

Reply:
Ok. So then we think it's unlikely that there will be conflicts?

Reply:
No risk of conflict, because the ID is unique per Freebox and it's also possible to add a custom one if you wish. So `freebox_1.freeboxos.fr`, `freebox_2.freeboxos.fr` ...

And for now, it's not possible to add two Freebox integrations 😛 .

Reply:
Ok. Why isn't it possible to add two config entries?

Reply:
Because it wasn't the case before, and at the beginning I didn't think it was useful since we are talking of a router, and every people have only one router !

Is it useful to add multiple routers to HA ?
I think only for a very very specific case of people who maintain a bunch of routers ...


While writing those lines, I am thinking of a use case when 1 person have 2 homes (the 1st the usual and connected one, the 2nd the vacation's one). Maybe the user would like to monitor it's vacation home to know if someone there.

Also, since there is already a lot of think in the PR, I was thinking of doing quickly after the merge.

Reply:
The user can add as many config entries as the user likes via the GUI, right?

Reply:
Emmm, yes, that was while importing that wasn't possible, just fixed this.

So now many config from import or UI, with unique DNS id.